web/src/user_groups.ts

```diff
@@ -202,6 +202,23 @@ export function get_recursive_subgroups(target_user_group: UserGroup): Set<numbe
     return subgroup_ids;
 }
 
+export function get_recursive_group_members(target_user_group: UserGroup): Set<number> {
+    const members = new Set(target_user_group.members);
+    const subgroup_ids = get_recursive_subgroups(target_user_group);
+
+    if (subgroup_ids === undefined) {
+        return members;
+    }
+
+    for (const subgroup_id of subgroup_ids) {
+        const subgroup = user_group_by_id_dict.get(subgroup_id)!;
```

Comment:
An assert would probably be most appropriate for this scenario.

Reply:
Fixed [here](https://github.com/zulip/zulip/compare/5a262b5339d4eb13585ab978b2caf053cf2e05d5..748e2d8c97be19d56506cd0057924da29a0ee134). But I'm curious to know why an `assert` statement over `!`, is it because ! checks for both null and undefined and we want to check only for undefined here?